lib/report/generateReport.ts

```diff
@@ -0,0 +1,198 @@
+import * as fs from "fs";
+import * as path from "path";
+import * as Mustache from "mustache";
+import {
+  OperationCoverageInfo,
+  TrafficValidationIssue,
+} from "../swaggerValidator/trafficValidator";
+
+export interface TrafficValidationIssueForRendering extends TrafficValidationIssue {
+  payloadFileLinkLabel?: string;
+}
+
+export interface OperationCoverageInfoForRendering extends OperationCoverageInfo {
+  specLinkLabel?: string;
+}
+
+// used to pass data to the template rendering engine
+export class CoverageView {
+  public package: string;
+  public language: string;
+  public apiVersion: string = "unknown";
+  public generatedDate: Date;  
+  
+  public undefinedOperationCount: number = 0;
+  public operationValidated: number = 0;
+  public operationFailed: number = 0;
+  public operationUnValidated: number = 0;  
+  public generalErrorResults: Map<string, TrafficValidationIssue[]>;
+
+  public validationResultsForRendering: TrafficValidationIssueForRendering[] = [];
+  public coverageResultsForRendering: OperationCoverageInfoForRendering[] = [];
+
+  private validationResults: TrafficValidationIssue[];  
+  private sortedValidationResults: TrafficValidationIssue[];
+  private coverageResults: OperationCoverageInfo[];
+
+  public constructor(
+    validationResults: TrafficValidationIssue[],
+    coverageResults: OperationCoverageInfo[],
+    undefinedOperationCount: number = 0,
+    packageName: string = "",
+    language: string = ""
+  ) {
+    this.package = packageName;
+    this.validationResults = validationResults;
+    this.coverageResults = coverageResults;
+    this.undefinedOperationCount = undefinedOperationCount;
+    this.generatedDate = new Date();
+    this.language = language;
+    this.generalErrorResults = new Map();
+    this.setMetrics();
+    this.sortOperationIds();
+    this.prepareDataForRendering();
+  }
+
+  private prepareDataForRendering() {
+    const specLinkPrefix = "https://github.com/Azure/azure-rest-api-specs/blob/main/";
+    const payloadLinkPrefix = "https://github.com/scbedd/oav-traffic-converter/blob/main/sample-tables-input/"
```

Comment:
Why hard code this here? can this be a argument?

Reply:
Exposed this command line parameter.